### DISCUSSION TITLE
Open File Dialogue

### DISCUSSION THREAD

**Discussion by ptstevens** [TomSchimansky/CustomTkinter#360] [Q&A] [answered]
Hi I have an existing Tkinter gui I am trying to smarten up with customeTkinter.

I have an open filedialogue function that returns a file name - in the old tkinter I was able to direct this to a label to show which file they have selected but cannot for love nor money get it to work here - any ideas?

**Comment by abhaymakes** [comment]
I dont exactly understand what you mean, do you want to add file dialogue in you tkinter app with customtkinter?

**Reply by ptstevens** [reply]
Sorry no open file dialog opens fine. Select a file ...this is fine. I need the name of the file to then be shown in a customtkinter label.

**Comment by ptstevens** [comment, marked as answer]
I have managed to work this out using
def button_output(self): self.entry_output.delete(0, 'end') file_name = fd.askopenfilename() filename = os.path.basename(str(file_name)) self.entry_output.insert(0, str(filename)) pass
